### PR TITLE
1D Column Examples documentation for Heat and Advection Equations

### PR DESCRIPTION
**Pull request by onzerem**
Hello, I was interested in adding some examples to the documentation. This PR adds the 1D column examples for the heat (heat.jl) and advection (advect.jl) equations to the documentation (under examples). For reference, this contribution was also discussed in Issues #2234 and #2242. 

Please let me know what you think of my contribution, and I can make any necessary edits. 

Thank you!

 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x]  Unit tests are included OR N/A.
- [x]  Code is exercised in an integration test OR N/A.
- [x]  Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by onzerem** [comment]
@Sbozzolo @dennisYatunin  Hello, I have one more pending check for my PR. I want to make sure my contributions build properly in the documentation, but I believe I may need admin approval in order for the buildkite/climacore-ci to run. I was also hoping for any feedback on my PR as well, I am more than happy to make any edits. Thank you!

**Comment by valeriabarra** [comment]
Hi, 
 why don't we see the `documenter / deploy` check preview? 
 
image


cc: @dennisYatunin , @Sbozzolo

**Comment by Sbozzolo** [comment]
> Hi, why don't we see the `documenter / deploy` check preview?
> 
> image
> 
> cc: @dennisYatunin , @Sbozzolo

Looking at the build look, I see this:
```
┌ Info: Deployment criteria for deploying preview build from GitHub Actions:
│ - ✔ ENV["GITHUB_REPOSITORY"]="CliMA/ClimaCore.jl" occurs in repo="github.com/CliMA/ClimaCore.jl.git"
│ - ✔ ENV["GITHUB_REF"] corresponds to a PR number
│ - ✘ PR originates from the same repository
│ - ✔ `push_preview` keyword argument to deploydocs is `true`
│ - ✔ ENV["GITHUB_ACTOR"] exists and is non-empty
│ - ✔ ENV["GITHUB_TOKEN"] exists and is non-empty
└ Deploying: ✘
```

And now that I see this, I think this makes sense: allowing third-party contributors to commit directly to a branch in the repo sounds like a security risk.

**Comment by valeriabarra** [comment]
> > Hi, why don't we see the `documenter / deploy` check preview?
> > image
> > cc: @dennisYatunin , @Sbozzolo
> 
> Looking at the build look, I see this:
> 
> ```
> ┌ Info: Deployment criteria for deploying preview build from GitHub Actions:
> │ - ✔ ENV["GITHUB_REPOSITORY"]="CliMA/ClimaCore.jl" occurs in repo="github.com/CliMA/ClimaCore.jl.git"
> │ - ✔ ENV["GITHUB_REF"] corresponds to a PR number
> │ - ✘ PR originates from the same repository
> │ - ✔ `push_preview` keyword argument to deploydocs is `true`
> │ - ✔ ENV["GITHUB_ACTOR"] exists and is non-empty
> │ - ✔ ENV["GITHUB_TOKEN"] exists and is non-empty
> └ Deploying: ✘
> ```
> 
> And now that I see this, I think this makes sense: allowing third-party contributors to commit directly to a branch in the repo sounds like a security risk.

HI, @Sbozzolo , thank you! 
Yes, I figured it was a security setting, indeed. So the only way to test documentation contributions for non-org members is to test locally on their machine, by running `docs/make.jl` and make sure that everything looks good locally, then?

**Comment by onzerem** [comment]
Screenshot (285)
Screenshot (287)

I was able to build my contribution locally. I added the screenshots of what it looks in the documentation. I'm open to any feedback for modifications. Thank you!
@Sbozzolo

File: docs/src/examples.md [hunk 1]
```diff
@@ -1,6 +1,84 @@
...
+\begin{equation}
+  \frac{\partial \theta}{\partial t} = -\frac{\partial (v \theta)}{\partial z}
+\label{eq:1d-column-advection-continuity}
+\end{equation}
+```
+This is discretized using the following
+
+```math
+\begin{equation}
+  \frac{\partial \theta}{\partial t} \approx - D(V, \theta)
```
Comment (valeriabarra) [inline docs/src/examples.md:21]: Hi @onzerem , here on line 21, you used capital `V`, but below, on line 29 you used lower case `v`. Please be consistent.

File: docs/src/examples.md [hunk 2]
```diff
@@ -1,6 +1,84 @@
...
+#### Prognostic Variables
+
+* ``\theta``: the scalar field
+* ``v``: the velocity field
+
+#### Tendencies 
+
+The example code solves the equation for 4 different tendencies with the following discretizations:
+
+* Tendency 1: ``D = \partial(UB),`` where ``\partial`` is the [`face-to-center divergence`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) and $UB$ is the [`center-to-face upwind product operator`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)
```
Comment (valeriabarra) [inline docs/src/examples.md:35]:
```suggestion
* Tendency 1: ``D = \partial(UB),`` where ``\partial`` is the [`face-to-center divergence`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) and $UB$ is the [`center-to-face upwind biased product`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.UpwindBiasedProductC2F) operator
```

File: docs/src/examples.md [hunk 3]
```diff
@@ -1,6 +1,84 @@
...
+* Tendenct 2: Follows tendency 1 with the addition of flux correction `fcc`
```
Comment (valeriabarra) [inline docs/src/examples.md:36]: Please describe in the formula and put a link to the `FluxCorrectionC2C` operator as you did for Tendency 1 here

File: docs/src/examples.md [hunk 4]
```diff
@@ -1,6 +1,84 @@
...
+* Tendency 3: $D = A$, where $A$ is the [`discrete vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C)
```
Comment (valeriabarra) [inline docs/src/examples.md:37]:
```suggestion
* Tendency 3: $D = A$, where $A$ is the [`center-to-center vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C) operator
```

File: docs/src/examples.md [hunk 5]
```diff
@@ -1,6 +1,84 @@
...
+* Tendency 4: Follows tendency 3 with the addition of flux correction `fcc`
```
Comment (valeriabarra) [inline docs/src/examples.md:38]: Same comment as for Tendency 2. Please spell out the formula and put a link for the operator used.

File: docs/src/examples.md [hunk 6]
```diff
@@ -1,6 +1,84 @@
...
+
+This test case is set up in a 1D column domain ``z \in [0, 4\pi]``. 
+
+### Heat
+
+The 1D Column heat example in [`examples/column/heat.jl`](https://github.com/CliMA/ClimaCore.jl/blob/main/examples/column/heat.jl).
+
+#### Equations and discretizations
+
+Follows the continuity equation
```
Comment (valeriabarra) [inline docs/src/examples.md:50]:
```suggestion
Follows the heat equation
```

File: docs/src/examples.md [hunk 7]
```diff
@@ -1,6 +1,84 @@
...
+```
+
+#### Prognostic Variables
+
+* ``\alpha``: thermal diffusivity measured in  $\frac{m^2}{s}$
+* ``T``: temperature
+
+#### Differentiation Operators
+
+ * ``D`` is the [face-to-center divergence](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C), called `divf2c` in the example code
```
Comment (valeriabarra) [inline docs/src/examples.md:75]:
```suggestion
 * ``D`` is the [face-to-center divergence](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) operator, called `divf2c` in the example code
```

File: docs/src/examples.md [hunk 8]
```diff
@@ -1,6 +1,84 @@
...
+ * ``G`` is the [center-to-face gradient](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.GradientC2F), called `gradc2f` in the example code
```
Comment (valeriabarra) [inline docs/src/examples.md:76]:
```suggestion
 * ``G`` is the [center-to-face gradient](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.GradientC2F) operator, called `gradc2f` in the example code
```

File: docs/src/examples.md [hunk 9]
```diff
@@ -1,6 +1,84 @@
...
+
+#### Set Up
+
+This test case is set up in a 1D column domain ``z \in [0, 1]``.
```
Comment (valeriabarra) [inline docs/src/examples.md:80]: Please also mention the boundary conditions. 

From the code
```
bcs_bottom = Operators.SetValue(FT(0.0))
bcs_top = Operators.SetGradient(Geometry.WVector(FT(1.0)))
```

we can see that it has a Dirichelet (homogeneous) BC at the bottom and a Neumann boundary condition at the top. 
I believe. Is this correct, @dennisYatunin ?

File: docs/src/examples.md [hunk 10]
```diff
@@ -1,6 +1,84 @@
 # Examples
 
-## 1D Column examples
+### Advection
+
+The 1D Column advection example in [`examples/column/advect.jl`](https://github.com/CliMA/ClimaCore.jl/blob/main/examples/column/advect.jl).
+
+#### Equations and Discretizations
+
+Follows the continuity equation
```
Comment (valeriabarra) [inline docs/src/examples.md:9]:
```suggestion
Follows the advection equation
```

File: docs/src/examples.md [hunk 11]
```diff
@@ -1,6 +1,84 @@
...
+The example code solves the equation for 4 different tendencies with the following discretizations:
+
+* Tendency 1: ``D = \partial(UB),`` where ``\partial`` is the [`face-to-center divergence`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) and $UB$ is the [`center-to-face upwind product operator`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.UpwindBiasedProductC2F)
+* Tendenct 2: Follows tendency 1 with the addition of flux correction `fcc`
+* Tendency 3: $D = A$, where $A$ is the [`discrete vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C)
+* Tendency 4: Follows tendency 3 with the addition of flux correction `fcc`
+
+#### Set Up
+
+This test case is set up in a 1D column domain ``z \in [0, 4\pi]``.
```
Comment (valeriabarra) [inline docs/src/examples.md:42]: You should probably mention the boundary conditions here and the form of velocity field used, which is a sinusoidal wave. Thank you

**Comment by onzerem** [comment]
Hi @valeriabarra, I added the suggestions. For the `FluxCorrectionC2C`, I was not able to find a link to that in the documentation, so I linked the source code. Let me know what you about the other edits. Thanks! 
Screenshot (298)
Screenshot (297)

File: docs/src/examples.md [hunk 12]
```diff
@@ -1,6 +1,85 @@
 # Examples
 
-## 1D Column examples
```
Comment (valeriabarra) [inline docs/src/examples.md:3]: Please put this header back in. 

```
## 1D Column examples
```

File: docs/src/examples.md [hunk 13]
```diff
@@ -1,6 +1,85 @@
...
+
+The 1D Column advection example in [`examples/column/advect.jl`](https://github.com/CliMA/ClimaCore.jl/blob/main/examples/column/advect.jl).
+
+#### Equations and Discretizations
+
+Follows the advection equation
+
+```math
+\begin{equation}
+  \frac{\partial \theta}{\partial t} = -\frac{\partial (v \theta)}{\partial z}
```
Comment (valeriabarra) [inline docs/src/examples.md:13]: Please use proper punctuation to connect or end sentences, even after math formulas/equations.

```suggestion
  \frac{\partial \theta}{\partial t} = -\frac{\partial (v \theta)}{\partial z} .
```

File: docs/src/examples.md [hunk 14]
```diff
@@ -1,6 +1,85 @@
...
+* Tendency 2: Follows tendency 1, where ``D = \partial(UB)``. Similarly, ``\partial`` is the [`face-to-center divergence`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) and $UB$ is the [`center-to-face upwind biased product`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.UpwindBiasedProductC2F) operator. Tendency 2 also includes the [`center-to-center flux correction`](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Operators/finitedifference.jl#L2617), `fcc`.
+* Tendency 3: $D = A$, where $A$ is the [`center-to-center vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C) operator.
+* Tendency 4: Follows tendency 3, where $D = A$, the [`center-to-center vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C). Tendency 4 also includes the [`center-to-center flux correction`](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Operators/finitedifference.jl#L2617), `fcc`.
```
Comment (valeriabarra) [inline docs/src/examples.md:38]: I would reorganize this a bit better.

```suggestion
- Tendency 1:

  $$D = \partial(UB),$$  

  where ``\partial`` is the [`face-to-center divergence`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.DivergenceF2C) and $UB$ is the [`center-to-face upwind biased product`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.UpwindBiasedProductC2F) operator.
- Tendency 2:

  $$D = \partial(UB) + \textrm{fcc}(v, \theta),$$  
  
  where $\textrm{fcc}(v, \theta)$ is the [`center-to-center flux correction`](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Operators/finitedifference.jl#L2617) operator.

- Tendency 3:

  $$D = A,$$  
   
  where $A$ is the [`center-to-center vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C) operator.

- Tendency 4:

  $$D = A + \textrm{fcc}(v, \theta),$$  
  
  where $\textrm{fcc}(v, \theta)$ is the [`center-to-center flux correction`](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Operators/finitedifference.jl#L2617) operator.
```

**Comment by valeriabarra** [comment]
> Hi @valeriabarra, I added the suggestions. For the `FluxCorrectionC2C`, I was not able to find a link to that in the documentation, so I linked the source code. Let me know what you about the other edits. Thanks! 

Hi @onzerem , it's fine to link directly to the source code, if the API documentation for that operator is missing. 

I left further suggestions. Also, given that the Heat Equation example is the simpler of the two (does not have multiple tendency specifications), I would put that first. Thank you.

File: docs/src/examples.md [hunk 15]
```diff
@@ -1,6 +1,85 @@
 # Examples
 
-## 1D Column examples
+### Advection
```
Comment (valeriabarra) [inline docs/src/examples.md:3]:
```suggestion
### Advection equation
```

File: docs/src/examples.md [hunk 16]
```diff
@@ -1,6 +1,85 @@
...
+* Tendency 4: Follows tendency 3, where $D = A$, the [`center-to-center vertical advection`](https://clima.github.io/ClimaCore.jl/dev/operators/#ClimaCore.Operators.AdvectionC2C). Tendency 4 also includes the [`center-to-center flux correction`](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Operators/finitedifference.jl#L2617), `fcc`.
+
+#### Set Up
+
+This test case is set up in a 1D column domain ``z \in [0, 4\pi]``, discretized into a mesh of 128 elements. The velocity field is defined as a sinusoidal wave. The boundary conditions are operator dependent, so they depend on the tendency. 
+* For tendencies 1 and 2 where the upwind biased operator ``UB`` is used, the left boundary is defined as ``sin(a - t)``. The right boundary is ``sin(b - t)``. Here ``a`` and ``b`` are the left and right bounds of the domain. 
+* For tendencies 3 and 4, where the advection operator ``A`` is used, the left boundary is defined as ``sin(-t)``.  The right boundary is extrapolated, meaning its value is set to the closest interior point.
+
+### Heat
```
Comment (valeriabarra) [inline docs/src/examples.md:46]:
```suggestion
### Heat equation
```

**Comment by onzerem** [comment]
Screenshot (312)
Screenshot (313)
Screenshot (314)

Here is another screenshot updated with the suggestions.

**Review by valeriabarra — APPROVED** [review]

**Comment by valeriabarra** [comment]
Hi @onzerem , thank you again for your contributiuon! Your PR looks good to me now and I approve it. 

You can please reduce the number of commits to une (using `git rebase -i` and then `git push -f`). If you have any questions about the rebasing process, please let me know. After you're done with that, we will merge your PR. Thank you again!
